consult.el

```diff
@@ -1183,45 +1183,70 @@ for which the command history is used."
    ;; If pressing "C-x M-:", i.e., `repeat-complex-command',
    ;; we are instead querying the `command-history' and get a full s-expression.
    ((eq last-command 'repeat-complex-command)
-    (delete-dups (mapcar #'prin1-to-string command-history)))
+    (cons (mapcar #'prin1-to-string command-history)
+          'minibuffer-history-position))
    ;; In the minibuffer we use the current minibuffer history,
    ;; which can be configured by setting `minibuffer-history-variable'.
    ((minibufferp)
-    (if (fboundp 'minibuffer-history-value)
-        (minibuffer-history-value) ;; Emacs 27 only
-      (symbol-value minibuffer-history-variable)))
+    (cons (if (fboundp 'minibuffer-history-value)
+              (minibuffer-history-value) ;; Emacs 27 only
+            (symbol-value minibuffer-history-variable))
+          'minibuffer-history-position))
```

Comment:
I am not sure if using the minibuffer-history-position is correct here, since it depends on the minibuffer-history-variable.